internal/controller/utils.go

```diff
@@ -1368,7 +1339,12 @@ func ProcessIngressClassParameters(tctx *provider.TranslateContext, c client.Cli
 	return nil
 }
 
-func GetIngressClass(ctx context.Context, c client.Client, log logr.Logger, ingressClassName string) (*networkingv1.IngressClass, error) {
+func FindMatchingIngressClass(ctx context.Context, c client.Client, log logr.Logger, obj client.Object) (*networkingv1.IngressClass, error) {
+	ingressClassName := ExtractIngressClass(obj)
+	return FindMatchingIngressClassByName(ctx, c, log, ingressClassName)
+}
+
+func FindMatchingIngressClassByName(ctx context.Context, c client.Client, log logr.Logger, ingressClassName string) (*networkingv1.IngressClass, error) {
```

Comment:
This function does not handle the request failure caused by network failure. This will cause unintented delete behavior.

Reply:
IngressClass is the main watch object, and all behavior is based on local caching, so it does not generate any network calls.